src/dbnode/encoding/m3tsz/timestamp_encoder.go

```diff
@@ -205,16 +205,15 @@ func (enc *TimestampEncoder) writeDeltaOfDeltaTimeUnitUnchanged(
 	return nil
 }
 
-func initialTimeUnit(start time.Time, tu xtime.Unit) xtime.Unit {
+func initialTimeUnit(start xtime.UnixNano, tu xtime.Unit) xtime.Unit {
 	tv, err := tu.Value()
 	if err != nil {
 		return xtime.None
 	}
 	// If we want to use tu as the time unit for start, start must
 	// be a multiple of tu.
-	startInNano := xtime.ToNormalizedTime(start, time.Nanosecond)
-	tvInNano := xtime.ToNormalizedDuration(tv, time.Nanosecond)
-	if startInNano%tvInNano == 0 {
+	tvInNano := xtime.UnixNano(xtime.ToNormalizedDuration(tv, time.Nanosecond))
```

Comment:
I think there may be an unnecessary conversion here

Reply:
Yeah you're right - since this is always nanoseconds we can just convert directly to the `UnixNano`